Issue #1729 Making sure generated images gets removed in clean target

hferentschik opened the pull request:
Addresses #1729 

This pull request tries to dynamically deal with cleaning up generated ditaa images (without having to explicitly list them). It also makes sure that generating the tarball for openshift.org gets build from a clean source directory. 

Whether the existence of the image in the tarball is really the reason for failure we are seeing I am not sure off, but this change makes sense regardless.

budhrg commented on centos_ci.sh:
```diff
@@ -266,7 +266,7 @@ function perform_release() {
   MINISHIFT_VM_DRIVER=kvm make integration_all
   exit_on_failure "$?" "Integration tests failed."
 
-  make link_check_docs IMAGE_UID=$(id -u) # Test docs builds and all links are valid
```
@LalatenduMohanty https://github.com/minishift/minishift/pull/1641 Not required anymore.

LalatenduMohanty replied:
@budhrg right, what I meant was that, this change has nothing to do with the original issue for the PR.

hferentschik commented:
@budhrg ,

> We don't need clean as part of this line now

Good catch. I removed the 'clean' from the command being run, but left it as target dependency. That makes it more explicit. 

Rebased as well. I think we should be able to merge now.

hferentschik commented:
I think this change will avoid the problem we see on the nightly openshift.org build by side effect. With this pull request, the generated tarfile will not contain any generated sources/files. 

Before we had the case that we actually included the generated PNG. My guess is that in this case the build tries to delete/update the image, but due to permission issues cannot do it.  Theoretically, there is really no problem having the generated image in the tarball. We can see that as well when doing the openshift.org doc build locally. I can successfully build the overall docs w/ or w/o the generated image being part of the tarball.

Since having a "clean" tarball w/o any generated files is nicer anyway, we should be good with this change. We can create an issue as part of the OpenShift docs to see whether this is something they want to look at from there end.

hferentschik commented:
@budhrg, I also split the commits now to make clear that the centos_ci.sh changes are unrelated.

budhrg commented:
Working fine for me :+1: 

```
$ make gen_adoc_tar
cd docs && docker run -u 1001 -e OPENSHIFT_VERSION=v3.6.0 -e MINISHIFT_VERSION=1.9.0 -e CENTOS_ISO_VERSION=v1.3.0 -e MINIKUBE_ISO_VERSION=v0.24.0 -tiv /home/budhram/gowork/src/github.com/minishift/minishift/docs:/minishift-docs:Z minishift/minishift-docs-builder:1.0.0 clean
rm -f source/command-ref/minishift.adoc source/command-ref/minishift_addons.adoc [..other adoc files..] source/command-ref/minishift_status.adoc source/command-ref/minishift_stop.adoc source/command-ref/minishift_update.adoc source/command-ref/minishift_version.adoc
rm -rf build
rm -f source/variables.adoc source/contributing/variables.adoc source/getting-started/variables.adoc source/openshift/variables.adoc source/troubleshooting/variables.adoc source/using/variables.adoc
******************ditaa images removal*********************
rm -f source/using/images/minishift-architecture.png.cache source/using/images/minishift-architecture.png
***************************************
cd docs && docker run -u 1001 -e OPENSHIFT_VERSION=v3.6.0 -e MINISHIFT_VERSION=1.9.0 -e CENTOS_ISO_VERSION=v1.3.0 -e MINIKUBE_ISO_VERSION=v0.24.0 -tiv /home/budhram/gowork/src/github.com/minishift/minishift/docs:/minishift-docs:Z minishift/minishift-docs-builder:1.0.0 adoc_tar
mkdir -p build
cp source/_topic_map.yml build/_topic_map.yml
pandoc --atx-headers --to=asciidoc --output=source/command-ref/minishift.adoc source/_tmp/minishift.md
[...]
```
